src/charts/CytoViz/components/DetailsLog/DetailsLog.js

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import useStyles from './style';
+
+function prettyPercentage(val) {
+  return (val * 100).toFixed(1) + '%';
+}
+
+function hiddenElements(cytoElements) {
+  let hiddenElements = 0;
+  cytoElements.forEach((element) => {
+    if (element.data('hidden')) {
+      hiddenElements++;
+    }
+  });
+  return hiddenElements;
+}
+
+const DetailsLog = (props) => {
+  const { cytoAsState } = props;
```

Comment:
To implement this HUD, which updates every time when the Cytoscape instance changes, I had to store the instance in a state variable, since the ref.current variable does not emit an event to rerender other components.